src/com/dabomstew/pkrandom/romhandlers/Gen7RomHandler.java

```diff
@@ -3196,4 +3198,92 @@ public AreaData() {
     public List<Integer> getAllHeldItems() {
         return Gen7Constants.allHeldItems;
     }
+
+    @Override
+    public List<Integer> getSensibleHeldItemsFor(TrainerPokemon tp, boolean consumableOnly, List<Move> moves, Map<Integer, List<MoveLearnt>> movesets) {
+        List<Integer> items = new ArrayList<>();
+        items.addAll(Gen7Constants.generalPurposeConsumableItems);
+        int frequencyBoostCount = 6; // Make some very good items more common, but not too common
+        if (!consumableOnly) {
+            frequencyBoostCount = 8; // bigger to account for larger item pool.
+            items.addAll(Gen7Constants.generalPurposeItems);
+        }
+        int[] pokeMoves = RomFunctions.getMovesAtLevel(tp.pokemon.number, movesets, tp.level);
+        boolean hasAtLeastOneDamagingMove = false;
+        for (int moveIdx : pokeMoves) {
+            Move move = moves.get(moveIdx);
+            if (move == null) {
+                continue;
+            }
+            if (move.category == MoveCategory.PHYSICAL) {
+                hasAtLeastOneDamagingMove = true;
+                items.add(Gen4Constants.liechiBerry);
+                items.add(Gen7Constants.consumableTypeBoostingItems.get(move.type));
+                if (!consumableOnly) {
+                    items.addAll(Gen7Constants.typeBoostingItems.get(move.type));
+                    items.add(Gen4Constants.choiceBand);
+                    items.add(Gen4Constants.muscleBand);
+                }
+            }
+            if (move.category == MoveCategory.SPECIAL) {
+                hasAtLeastOneDamagingMove = true;
+                items.add(Gen4Constants.petayaBerry);
+                items.add(Gen7Constants.consumableTypeBoostingItems.get(move.type));
+                if (!consumableOnly) {
+                    items.addAll(Gen7Constants.typeBoostingItems.get(move.type));
+                    items.add(Gen4Constants.wiseGlasses);
+                    items.add(Gen4Constants.choiceSpecs);
+                }
+            }
+            if (!consumableOnly && Gen7Constants.moveBoostingItems.containsKey(moveIdx)) {
+                items.addAll(Gen7Constants.moveBoostingItems.get(moveIdx));
+            }
+        }
+        if (hasAtLeastOneDamagingMove) {
```

Comment:
Same thing as what I said in Gen6RomHandler about Assault Vest

Reply:
Done.